app-editors/qhexedit2/qhexedit2-0.8.10.ebuild

```diff
@@ -0,0 +1,144 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+inherit python-r1 qmake-utils
+
+DESCRIPTION="Hex editor library, Qt application written in C++ with Python bindings"
+HOMEPAGE="https://github.com/Simsys/qhexedit2/"
+SRC_URI="https://github.com/Simsys/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+S="${WORKDIR}/${PN}-${PV}"
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~riscv ~x86"
+IUSE="doc +gui python qt5 +qt6"
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )
+	|| ( qt5 qt6 )
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.8.10-pyqt6.patch"
+)
+
+RDEPEND="
+	media-libs/libglvnd
+	python? (
+		${PYTHON_DEPS}
+		qt5? (
+			$(python_gen_cond_dep '
+				>=dev-python/pyqt5-5.15.6[gui,widgets,${PYTHON_USEDEP}]
+				>=dev-python/pyqt5-sip-12.9:=[${PYTHON_USEDEP}]
+			')
+		)
+		qt6? (
+			$(python_gen_cond_dep '
+				>=dev-python/pyqt6-6.8.0[gui,widgets,${PYTHON_USEDEP}]
+				>=dev-python/pyqt6-sip-13.5:=[${PYTHON_USEDEP}]
+			')
+		)
+	)
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+	)
+	qt6? (
+		dev-qt/qtbase:6[gui,widgets]
+	)
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	python? (
+		$(python_gen_cond_dep '
+			>=dev-python/pyqt-builder-1.10[${PYTHON_USEDEP}]
+			>=dev-python/sip-5:=[${PYTHON_USEDEP}]
+		')
+	)
+"
+
+src_prepare() {
+	default
+	sed -i -e '/^unix:DESTDIR/ d' -e "\$atarget.path = /usr/$(get_libdir)" \
+		-e "\$aINSTALLS += target" src/qhexedit.pro \
+		|| die "src/qhexedit.pro: sed failed"
+	sed -i '/abi-version/d' pyproject.toml \
+		|| die "pyproject.toml: sed failed"
+	if use qt6; then
+		sed -i -e 's/PyQt5/PyQt6/g' pyproject.toml project.py setup.py \
+			|| die "pyproject.toml project.py setup.py: sed failed"
+	elif use qt5; then
+		sed -i -e 's/PyQt6/PyQt5/g' pyproject.toml project.py setup.py \
+			|| die "pyproject.toml project.py setup.py: sed failed"
+	fi
+}
+
+src_configure() {
+	if use qt6; then
+		eqmake6 src/qhexedit.pro
+	elif use qt5; then
+		eqmake5 src/qhexedit.pro
+	fi
+
+	if use gui; then
+		cd example || die "can't cd example"
+		if use qt6; then
+			eqmake6 qhexedit.pro
+		elif use qt5; then
+			eqmake5 qhexedit.pro
+		fi
+	fi
```

Comment:
That sounds like IUSE `gui` should maybe just be `example`? Or is this something people are *actually* using as a hexeditor user interface?

Reply:
![Screenshot from 2025-01-24 14-14-10](https://github.com/user-attachments/assets/85a3ec3b-1716-49c2-9e23-63c9cb7176a6)

The `example` here are quite deceiving, but in fact there is a full-featured hex editor with a graphical interface, as shown in the figure.